pkg/container/vector/tools.go

```diff
@@ -15,60 +15,62 @@
 package vector
 
 import (
+	"reflect"
 	"strings"
-
-	"github.com/matrixorigin/matrixone/pkg/common/mpool"
+	"unsafe"
 
 	"github.com/matrixorigin/matrixone/pkg/common/moerr"
+	"github.com/matrixorigin/matrixone/pkg/common/mpool"
 	"github.com/matrixorigin/matrixone/pkg/container/nulls"
 	"github.com/matrixorigin/matrixone/pkg/container/types"
 	"github.com/matrixorigin/matrixone/pkg/pb/api"
 	"github.com/matrixorigin/matrixone/pkg/pb/plan"
 	"golang.org/x/exp/constraints"
 )
 
-// CheckInsertVector  vector.data will not be nil when insert rows
-func CheckInsertVector(v *Vector, m *mpool.MPool) *Vector {
-	if v.data == nil && v.Typ.Oid != types.T_any && v.Length() > 0 {
-		newVec := New(v.Typ)
-		newVec.isConst = v.isConst
-		val := GetInitConstVal(v.Typ)
-		for i := 0; i < v.Length(); i++ {
-			newVec.Append(val, true, m)
-		}
-		newVec.length = v.length
-		return newVec
-	}
-	return v
-}
-
-func MustTCols[T types.FixedSizeT](v *Vector) []T {
+func MustTCols[T any](v *Vector) []T {
 	// XXX hack.   Sometimes we generate an t_any, for untyped const null.
 	// This should be handled more carefully and gracefully.
 	if v.GetType().Oid == types.T_any {
 		return nil
 	}
-
-	if t, ok := v.Col.([]T); ok {
-		return t
+	if v.class == CONSTANT {
```

Comment:
Is it possible that we don't expand a constant vector to its length?

Reply:
Of course.... . vector can provide an iterator interface, and then everyone can use this interface. But then